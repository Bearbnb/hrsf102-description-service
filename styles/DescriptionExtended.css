.container {
  font-size: 16px;
  letter-spacing: normal;
  font-weight: normal;
  line-height: 22px;
}

.fadeIn {
  animation: fadeIn 0.5s ease-in both;
}

.fadeOut {
  display: none;
  animation: fadeOut 0.5s ease-in both;
}

@keyframes fadeIn {
	from {
		opacity: 0;
		transform: translate3d(0, -1%, 0);
	}
	to {
		opacity: 1;
		transform: translate3d(0, 0, 0);
	}
}
