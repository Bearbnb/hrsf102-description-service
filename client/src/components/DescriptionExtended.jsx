import React from 'react';
import PropTypes from 'prop-types';

import styles from '../../../styles/DescriptionExtended.css';

const DescriptionExtended = ({
  descriptionExtended,
  hideDescription,
  showDescriptionExtended,
}) => {
  const bullets = Object.keys(descriptionExtended);

  const fade = showDescriptionExtended ? styles.fadeIn : styles.fadeOut;

  return (
    <div className={`${styles.container} ${fade}`}>
      {bullets.map(bullet => (
        <div>
          <span>{bullet}</span>
          {descriptionExtended[bullet].map(paragraph => (
            <p>{paragraph}</p>
          ))}
        </div>
      ))}
      <button
        type="button"
        onClick={() => hideDescription()}
      >Hide
      </button>
    </div>
  );
};

DescriptionExtended.propTypes = {
  descriptionExtended: PropTypes.objectOf(PropTypes.string),
  hideDescription: PropTypes.func,
  showDescriptionExtended: PropTypes.bool,
};

DescriptionExtended.defaultProps = {
  descriptionExtended: {
    'THE SPACE': 'IT\'S LARGE',
    'VERY IMPORTANT:': 'DON\'T TOUCH THINGS',
    'GUEST ACCESS': 'USE THE DOOR',
  },
  hideDescription: () => {},
  showDescriptionExtended: false,
};

export default DescriptionExtended;
