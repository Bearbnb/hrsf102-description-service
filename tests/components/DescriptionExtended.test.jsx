import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import DescriptionExtended from '../../client/src/components/DescriptionExtended';

describe('<DescriptionExtended />', () => {
  it('should render properly', () => {
    const descriptionExtended = {
      'THE SPACE': ['THERE\'S A LOT'],
      'VERY IMPORTANT': ['TESTS ARE IMPORTANT'],
      'GUEST ACCESS': ['DRIVE THERE'],
    };

    const wrapper = shallow(<DescriptionExtended descriptionExtended={descriptionExtended} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
