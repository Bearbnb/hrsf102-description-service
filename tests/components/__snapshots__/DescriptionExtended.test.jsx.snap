// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DescriptionExtended /> should render properly 1`] = `
<div
  className="container fadeOut"
>
  <div>
    <span>
      THE SPACE
    </span>
    <p>
      THERE'S A LOT
    </p>
  </div>
  <div>
    <span>
      VERY IMPORTANT
    </span>
    <p>
      TESTS ARE IMPORTANT
    </p>
  </div>
  <div>
    <span>
      GUEST ACCESS
    </span>
    <p>
      DRIVE THERE
    </p>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Hide
  </button>
</div>
`;
